File: Java (client)/Persistence/src/main/java/ro/mpp2024/TeamDBRepository.java
package ro.mpp2024;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class TeamDBRepository implements ITeamRepository {

    private JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger(TeamDBRepository.class);

    public TeamDBRepository(Properties props){
        logger.info("Initializing TeamDBRepository with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }

    @Override
    public Team save(Team entity) {
        logger.traceEntry("saving team {} ",entity);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("insert into teams(name) values (?) ")){
            statement.setString(1, entity.getName());
            int result = statement.executeUpdate();
        }catch(SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return entity;
    }

    @Override
    public Team delete(Integer aLong) {
        return null;
    }

    @Override
    public Team update(Team entity) {
        return null;
    }

    @Override
    public Team findOne(Integer aLong) {
        logger.traceEntry("finding team with id {} ",aLong);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("select * from teams where id = ?")) {
            statement.setInt(1, aLong);
            try (var result = statement.executeQuery()) {
                if (result.next()) {
                    int id = result.getInt("id");
                    String name = result.getString("name");
                    Team team = new Team(id, name);
                    return team;
                }
            }
        }catch (SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return null;
    }

    @Override
    public Iterable<Team> findAll() {
        logger.traceEntry("finding all teams");
        List<Team> teams = new ArrayList<>();
        Connection con=dbUtils.getConnection();
        try(PreparedStatement statement = con.prepareStatement("select * from teams")){
            try(var result = statement.executeQuery()){
                while(result.next()){
                    int id = result.getInt("id");
                    String name = result.getString("name");
                    Team team = new Team(id, name);
                    teams.add(team);
                }
            }}
        catch (SQLException e){
            logger.error(e);
            System.out.println("Error DB "+e);
        }
        return teams;
    }

}

File: C# (server)/Persistence/TeamDBRepository.cs

using System;
using System.Collections.Generic;
using System.Data;
using Model;
using log4net;
using log4net.Util;

namespace Persistence
{
    public class TeamDBRepository : ITeamRepository
    {
        private static readonly ILog logger = LogManager.GetLogger("TeamDBRepository");

        IDictionary<String, string> props;

        public TeamDBRepository(IDictionary<String, string> props)
        {
            logger.Info("Creating TeamDBRepository");
            this.props = props;
        }

        public IEnumerable<Team> findAll()
        {
            logger.Info("Finding all teams");
            IList<Team> teams = new List<Team>();
            IDbConnection connection = DBUtils.getConnection(props);
            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select id, name from teams";
                using (var result = command.ExecuteReader())
                {
                    while (result.Read())
                    {
                        int id = result.GetInt32(0);
                        string name = result.GetString(1);
                        Team team = new Team(id, name);
                        teams.Add(team);
                        
                        teams.Add(team);
                    }
                }
            }

            logger.InfoExt(teams);
            return teams;
        }

        public Team findOne(int id)
        {
            logger.Info("Finding team with id " + id);
            IDbConnection connection = DBUtils.getConnection(props);
            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select id,name from teams where id=@id";
                IDbDataParameter paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                command.Parameters.Add(paramId);
                using (var result = command.ExecuteReader())
                {
                    if (result.Read())
                    {
                        int idv = result.GetInt32(0);
                        string name = result.GetString(1);
                        Team team = new Team(idv, name);
                        logger.InfoExt(team);
                        return team;
                    }
                }
            }

            return null;
        }

        public Team save(Team entity)
        {
            throw new System.NotImplementedException();
        }

        public Team delete(int id)
        {
            throw new System.NotImplementedException();
        }

        public Team update(Team entity)
        {
            throw new System.NotImplementedException();
        }
    }
}